web/src/test/java/org/cbioportal/genome_nexus/web/CancerHotspotsIntegrationTest.java

```diff
@@ -0,0 +1,191 @@
+
+        Hotspot[] hotspots5 = restTemplate.getForObject(
+            baseUrl + genomicLocations[5], Hotspot[].class);
+
+        // not a hotspot, there should me no match
+        assertEquals(0, hotspots5.length);
+
+        // genomic location instances (needed for POST requests)
+        List<GenomicLocation> genomicLocationsInstances = Arrays.stream(genomicLocations).map(
+            g -> this.notationConverter.parseGenomicLocation(g, ",")).collect(Collectors.toList());
+
+        // POST request
+        AggregatedHotspots[] aggregatedHotspots = restTemplate.postForObject(
+            baseUrl, genomicLocationsInstances, AggregatedHotspots[].class);
+
+        // for each genomic location we should have one matching AggregatedHotspots instance
+        assertEquals(genomicLocations.length, aggregatedHotspots.length);
+
+        // GET and POST requests should return exact same hotspots
+        assertEquals(aggregatedHotspots[0].getHotspots().size(), hotspots0.length);
+        assertEquals(aggregatedHotspots[1].getHotspots().size(), hotspots1.length);
+        assertEquals(aggregatedHotspots[2].getHotspots().size(), hotspots2.length);
+        assertEquals(aggregatedHotspots[3].getHotspots().size(), hotspots3.length);
+        assertEquals(aggregatedHotspots[4].getHotspots().size(), hotspots4.length);
+        assertEquals(aggregatedHotspots[5].getHotspots().size(), hotspots5.length);
+
+        // genomic locations should be mapped correctly for the POST request
+        assertEquals(this.notationConverter.genomicToHgvs(genomicLocationsInstances.get(0)),
+            this.notationConverter.genomicToHgvs(aggregatedHotspots[0].getGenomicLocation()));
+        assertEquals(this.notationConverter.genomicToHgvs(genomicLocationsInstances.get(1)),
+            this.notationConverter.genomicToHgvs(aggregatedHotspots[1].getGenomicLocation()));
+        assertEquals(this.notationConverter.genomicToHgvs(genomicLocationsInstances.get(2)),
+            this.notationConverter.genomicToHgvs(aggregatedHotspots[2].getGenomicLocation()));
+        assertEquals(this.notationConverter.genomicToHgvs(genomicLocationsInstances.get(3)),
+            this.notationConverter.genomicToHgvs(aggregatedHotspots[3].getGenomicLocation()));
+        assertEquals(this.notationConverter.genomicToHgvs(genomicLocationsInstances.get(4)),
+            this.notationConverter.genomicToHgvs(aggregatedHotspots[4].getGenomicLocation()));
+        assertEquals(this.notationConverter.genomicToHgvs(genomicLocationsInstances.get(5)),
+            this.notationConverter.genomicToHgvs(aggregatedHotspots[5].getGenomicLocation()));
+    }
+
+    @Test
+    public void testIndelHotspots()
+    {
+        String baseUrl = "http://localhost:38888/cancer_hotspots/genomic/";
+
+        // genomic location strings (needed for GET requests)
+        String[] genomicLocations = {
+            "7,55242466,55242480,GAATTAAGAGAAGCA,-",
+            "7,55249002,55249003,-,CAGCGTGGA"
```

Comment:
@onursumer this is great! Looks like we are using MAF format here. Maybe good to have some parameter indicating that. 

In the same vain - how are impossible ref/alt/position combos handled?

Reply:
ill just filed an issue for my comments so we can merge this.

https://github.com/genome-nexus/genome-nexus/issues/140